process/peer/process.go

```diff
@@ -379,6 +383,46 @@ func (vs *validatorStatistics) GetValidatorInfoForRootHash(rootHash []byte) (map
 	return vInfos, err
 }
 
+// ProcessRatingsEndOfEpoch makes end of epoch process on the rating
+func (vs *validatorStatistics) ProcessRatingsEndOfEpoch(validatorInfos map[uint32][]*state.ValidatorInfo) error {
+	if len(validatorInfos) == 0 {
+		return process.ErrNilValidatorInfos
+	}
+
+	for shardId, validators := range validatorInfos {
+		for _, validator := range validators {
+			validatorAppereances := validator.ValidatorSuccess + validator.ValidatorFailure
```

Comment:
this can be extracted in a function as it is hard to read and understand. for in for and if in if.

Reply:
fixed